spec/models/budget/investment_spec.rb

```diff
@@ -414,6 +414,47 @@
     end
   end
 
+  describe ".visible_to_valuator" do
+    let(:valuator) { create(:valuator) }
+
+    it "returns investments assigned to the valuator" do
+      investment = create(:budget_investment, :visible_to_valuators, valuators: [valuator])
+
+      expect(Budget::Investment.visible_to_valuator(valuator)).to eq [investment]
+    end
+
+    it "does not return investments assigned to other valuators" do
+      create(:budget_investment, :visible_to_valuators, valuators: [create(:valuator)])
+
+      expect(Budget::Investment.visible_to_valuator(valuator)).to be_empty
+    end
+
+    it "does not return duplicate investments when they're assigned more than once" do
+      investment = create(:budget_investment, :visible_to_valuators)
```

Comment:
```suggestion
      valuator_group = create(:valuator_group, valuators: [valuator])
      investment = create(:budget_investment, :visible_to_valuators, valuator_groups: [valuator_group])
```

With this change, we will also cover the valuator assignment through a group, or we can also a test for this scenario:

```
    it "does not return duplicate investments when valuator is directly assigned and belongs to the group" do
      valuator_group = create(:valuator_group, valuators: [valuator])
      investment = create(:budget_investment, :visible_to_valuators, valuators: [valuator],
                                                                     valuator_groups: [valuator_group])

      expect(Budget::Investment.visible_to_valuator(valuator)).to eq [investment]
    end
```

Reply:
@Senen I've added the test. Thanks! :smile: